src/components/Loading/Loading.tsx

```diff
@@ -0,0 +1,44 @@
+import Typography from '../Typography';
+import React, { useEffect, useState } from 'react';
+import './Loading.scss';
+
+export interface LoadingProps {
+  labels: string[];
+}
+
+export const Loading = (props: LoadingProps) => {
+  const { labels } = props;
+  const [currentLabel, setCurrentLabel] = useState(labels[0]);
+
+  let currentLabelIndex = 0;
+  const newLabel = () => {
+    if (currentLabelIndex >= labels.length - 1) return;
+    setTimeout(() => {
+      setCurrentLabel(labels[currentLabelIndex]);
+      currentLabelIndex++;
+      newLabel();
+    }, Math.max(3000, Math.random() * 6000));
+  };
+
+  useEffect(() => {
+    newLabel();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
This changes the label ever few seconds, useEffect starts it, then it gets called repeatedly until the end of the list of labels is reached.